src/workerd/api/sql-test.js

```diff
@@ -922,6 +1002,45 @@ async function testForeignKeys(storage) {
   }
 }
 
+async function testStreamingIngestion(request, storage) {
+  const { sql } = storage
+
+  sql.exec(`CREATE TABLE streaming(val TEXT);`)
+
+  const reader = request.body.getReader()
+  const decoder = new TextDecoder()
+
+  await storage.transaction(async () => {
+    let buffer = ''
+    while (true) {
+      const { value, done } = await reader.read()
```

Comment:
non-blocking... but just reminder.... Keep in mind that this is going to give you a new 4096-byte chunk allocated for each read. For reading a `request.body`, it's way more efficient to use a `byob` reader with a single allocated buffer up front.

Reply:
Moved to `TextDecoderStream` as suggested in chat.